### PR TITLE
feat: show linked issue type and id in listTestCases

### PR DESCRIPTION
**Pull request by SavioBS629**

### CODE REVIEW

File: src/tools/testmanagement-utils/list-testcases.ts [hunk 1]
```diff
@@ -85,10 +85,14 @@ export async function listTestCases(
 
     // Summary for more focused output
     const summary = test_cases
-      .map(
-        (tc: any) =>
-          `• ${tc.identifier}: ${tc.title} [${tc.case_type} | ${tc.priority}]`,
-      )
+      .map((tc: any) => {
+        const links = (tc.issues ?? [])
+          .map((i: any) => `${i.issue_type}:${i.jira_id}`)
```
Comment (SavioBS629) [inline src/tools/testmanagement-utils/list-testcases.ts:91]: **[Low] Unguarded issue fields can render `undefined:undefined`**

`links` is built as `${i.issue_type}:${i.jira_id}` without checking the fields exist. The outer `links ? …` guard only catches an *empty* array, not falsy field values — so a malformed entry like `issues: [{}]` would render `{linked: undefined:undefined}`.

**Suggestion:** guard per-issue before mapping, e.g. `.filter((i: any) => i?.issue_type && i?.jira_id)`. Low priority — only matters if the TM API doesn't guarantee these fields when `issues` is non-empty.

_Reviewer: stack:code-review_

**Review by SavioBS629 — COMMENTED** [review]
Claude Code Review (automated) — 1 inline finding(s). Full report in the PR comment below. Verdict: Passed.

**Comment by SavioBS629** [comment]
# Claude Code PR Review

**PR:** https://github.com/browserstack/mcp-server/pull/334  •  **Head:** be9bbb2  •  **Reviewers:** stack:code-review

## Summary
Extends the `listTestCases` summary line to append `{linked: <issue_type>:<jira_id>, ...}` when a test case has linked issues (e.g. Linear/Jira), and adds a vitest file covering the linked and no-issues cases.

## Review Table

| Priority | Category | Check | Status | Notes |
|----------|----------|-------|--------|-------|
| High | Security | No hardcoded secrets or credentials | Pass | No secrets touched |
| High | Security | Authentication/authorization checks present | N/A | Auth path unchanged |
| High | Security | Input validation and sanitization | Pass | `tc.issues ?? []` guards missing field |
| High | Security | No IDOR — resource ownership validated | N/A | No new resource access |
| High | Security | No SQL injection (parameterized queries) | N/A | No DB access |
| High | Correctness | Logic is correct, handles edge cases | Pass | Empty-array case handled & tested |
| High | Correctness | Error handling is explicit, no swallowed exceptions | Pass | Inside existing try/catch → `formatAxiosError` |
| High | Correctness | No race conditions or concurrency issues | N/A | Pure formatting |
| Medium | Testing | New code has corresponding tests | Pass | Two new vitest cases |
| Medium | Testing | Error paths and edge cases tested | Pass | No-issues case covered |
| Medium | Testing | Existing tests still pass (no regressions) | Pass | Formatting-only, backward compatible |
| Medium | Performance | No N+1 queries or unbounded data fetching | N/A | No new fetches |
| Medium | Performance | Long-running tasks use background jobs | N/A | — |
| Medium | Quality | Follows existing codebase patterns | Pass | Matches surrounding `(tc: any)` style |
| Medium | Quality | Changes are focused (single concern) | Pass | One feature, one test file |
| Low | Quality | Meaningful names, no dead code | Pass | — |
| Low | Quality | Comments explain why, not what | Pass | — |
| Low | Quality | No unnecessary dependencies added | Pass | None added |

## Findings

- **File:** `src/tools/testmanagement-utils/list-testcases.ts:90`
- **Severity:** Low
- **Reviewer:** stack:code-review
- **Issue:** `links` is built as `${i.issue_type}:${i.jira_id}` without validating those fields exist. The `links ? …` guard only checks for an *empty* array, not falsy field values, so a malformed issue entry (e.g. `issues: [{}]`) renders `{linked: undefined:undefined}`.
- **Suggestion:** Guard per-issue before mapping, e.g. `.filter((i: any) => i?.issue_type && i?.jira_id)`, so malformed entries are skipped rather than shown as `undefined:undefined`. Low priority — depends on whether the TM API guarantees these fields when `issues` is non-empty.

_Non-blocking note (not introduced by this PR): the field `jira_id` holds a Linear identifier for Linear-tracked issues, mirroring the upstream API field name — worth a follow-up only if the API renames it._

---

**Verdict: PASS** — focused, backward-compatible, well-tested change; one low-severity robustness nit.

**Comment by SavioBS629** [comment]
# Claude Code PR Review

**PR:** https://github.com/browserstack/mcp-server/pull/334  •  **Head:** 0f094a5  •  **Reviewers:** stack:code-review

_Continues [the previous review](https://github.com/browserstack/mcp-server/pull/334#issuecomment-4923095443) — changes since `be9bbb2` (delta)._

## Summary
Adds linked-issue labels (`{linked: <issue_type>:<jira_id>}`) to the `listTestCases` summary. This delta guards against malformed issue entries and adds a test for that case — resolving the one open finding from the previous review.

## Review Table

| Priority | Category | Check | Status | Notes |
|----------|----------|-------|--------|-------|
| High | Security | No hardcoded secrets or credentials | Pass | No secrets touched |
| High | Security | Authentication/authorization checks present | N/A | Auth path unchanged |
| High | Security | Input validation and sanitization | Pass | Malformed issue entries now filtered out |
| High | Security | No IDOR — resource ownership validated | N/A | No new resource access |
| High | Security | No SQL injection (parameterized queries) | N/A | No DB access |
| High | Correctness | Logic is correct, handles edge cases | Pass | Malformed/empty issue cases handled & tested |
| High | Correctness | Error handling is explicit, no swallowed exceptions | Pass | Inside existing try/catch → `formatAxiosError` |
| High | Correctness | No race conditions or concurrency issues | N/A | Pure formatting |
| Medium | Testing | New code has corresponding tests | Pass | Three vitest cases |
| Medium | Testing | Error paths and edge cases tested | Pass | Malformed-entry + no-issues cases covered |
| Medium | Testing | Existing tests still pass (no regressions) | Pass | Formatting-only, backward compatible |
| Medium | Performance | No N+1 queries or unbounded data fetching | N/A | No new fetches |
| Medium | Performance | Long-running tasks use background jobs | N/A | — |
| Medium | Quality | Follows existing codebase patterns | Pass | Matches surrounding `(i: any)` style |
| Medium | Quality | Changes are focused (single concern) | Pass | One feature, one test file |
| Low | Quality | Meaningful names, no dead code | Pass | — |
| Low | Quality | Comments explain why, not what | Pass | — |
| Low | Quality | No unnecessary dependencies added | Pass | None added |

## Findings

- ~~`src/tools/testmanagement-utils/list-testcases.ts:90` Low — Unguarded issue fields render `undefined:undefined`~~ **Resolved** — this delta adds `.filter((i: any) => i?.issue_type && i?.jira_id)` before the `.map`, so malformed entries are skipped, plus a new test ("skips malformed issue entries instead of rendering undefined") asserting the output contains `{linked: linear:TES-9}` and no `undefined`.

No open findings. The delta cleanly addresses the prior review with no new issues.

---

**Verdict: PASS** — prior low-severity nit fixed and covered by a new test; nothing outstanding.

**Review by ruturaj-browserstack — APPROVED** [review]
